src/interfaces/chain.cpp

```diff
@@ -346,13 +346,14 @@ class ChainImpl : public Chain
     {
         return MakeUnique<NotificationsHandlerImpl>(*this, notifications);
     }
-    void waitForNotificationsIfTipChanged(const uint256& old_tip) override
+    bool waitForNotificationsIfTipIsNotSame(const uint256& tip) override
```

Comment:
I like the name change, and I can see how the having the bool return value should make races and the deadlock reported https://github.com/bitcoin/bitcoin/issues/16307#issuecomment-595786198 less likely, though I suspect there is also probably a more direct and reliable fix we can find